tests/Unit/Repositories/ProfileRepositoryTest.php

```diff
@@ -254,4 +254,38 @@ public function profiles_should_be_grouped()
             $this->assertTrue(array_key_exists($item['display_name'], $profiles['profiles']));
         });
     }
+
+    /**
+     * @covers App\Repositories\ProfileRepository::getProfilesByGroupOrderPiped
+     * @test
+     */
+    public function profile_group_ids_should_return_ordered_array()
+    {
+        // Mock the user listing
+        $return_user_listing = app('Factories\Profile')->create(10);
+
+        $groups = collect($return_user_listing)->map(function ($item) {
+            return array_shift($item['groups']);
+        })->unique()->toArray();
+
+        krsort($groups);
```

Comment:
This can be done in the collection above?

Reply:
Chris and I spent over two hours getting to a "working" state. We know there's cleanup. The krsort here is because we went back and forth between collections and arrays a few times and were getting stuck, so we just left this as a "hey at least it works" state.